react/features/app/components/App.web.js

```diff
@@ -38,6 +41,31 @@ export class App extends AbstractApp {
     }
 
     /**
+     *  Sets handler for beforeunload event after mounting of the component.
+     *
+     *  @returns {void}
+     */
+    componentDidMount() {
+        // Stops collecting the logs and disposing the API when the user
+        // closes the page.
+        const store = this._getStore();
+        const {
+            logCollector,
+            logCollectorStarted
+        } = store.getState()['features/base/logger'];
+
+        window.addEventListener('beforeunload', () => {
+            // Stop the LogCollector
+            if (logCollectorStarted) {
+                logCollector.stop();
```

Comment:
No, please stop piling logic into features/app and the App component. You just defined feature/base/logger so use a middleware inside it to act on actions. If that absolutely need to happen in `componentDidMount`, then introduce an `APP_DID_MOUNT` action type and middleware it inside features/base/logger.